### ISSUE TITLE
arc deploy doesn't flag options @http mappings below '/' but arc sandbox does

### ISSUE THREAD

**Issue by krimple** [architect/architect#790]
Architect lets you put OPTIONS mappings against '/' so you can do CORS headers, etc. The '/' mapping is greedy; any calls to OPTIONS or other methods below that don't map something end up in the index lambda.

The `arc deploy` process deploys those lambdas, properly mapped via RESTful routes even to sub-paths with parameters.

The lambdas are getting executed.

However, running `arc sandbox` flags these methods.

Add mappings for http with OPTIONS deeper than root:

```
@http
get /
options /tasks/:taskOwner
options /tasks/:taskOwner/:taskId
options /tasks
...
```

2. Deploy them with `arc deploy`

3. Run the sandbox:

```
arc sandbox
---
kenrimple:architect-serverless-app/ (master*) $ [aws-venv] arc sandbox                                                                                                                             [13:03:23]
         App ⌁ architect-serverless-app
      Region ⌁ us-east-1
     Profile ⌁ default
     Version ⌁ Architect 6.3.5
         cwd ⌁ /Users/kenrimple/projects/serverless-schmerverless-ete-2020/architect-serverless-app

× Error: Architect schema validation errors
  @http invalid route: "options /tasks/:taskOwner"
  @http invalid route: "options /tasks/:taskOwner/:taskId"
  @http invalid route: "options /tasks"
```

**Expected behavior**

Either support in sandbox of lower-level OPTIONS paths or fail deploy so you can fix the routes that are invalid and not deploy code that isn't supported in the sandbox.

**Screenshots**
None

**Desktop**
 - OS: [e.g. iOS] OS X Catalina
 - Browser [e.g. chrome, safari] N/A
 - Version [e.g. 22] Architect 6.3.5

**Comment by ryanblock** [comment]
So you're saying this Arc file:
```
@http
get /
options /tasks/:taskOwner
options /tasks/:taskOwner/:taskId
options /tasks
```
With the following files in the project:
```
src/http/get-index/index.js
src/http/options-tasks-000taskOwner/index.js
src/http/options-tasks-000taskOwner-000taskId/index.js
src/http/options-tasks/index.js
```
Deploys to AWS, and if you were to make an `options` request to `{your URL}/tasks/foo/bar` the handler at `src/http/options-tasks-000taskOwner-000taskId/index.js` will be invoked?

I don't believe that is currently possible, and is the crux of this issue: https://github.com/architect/architect/issues/760

**Comment by krimple** [comment]
I am going to create an empty project to reproduce this for sure.

**Comment by krimple** [comment]
@ryanblock in terms of your question above, I might have tested one or two of the options URLs from the API Gateway tester, so it's _possible_ that they are being caught by the index.js normally. I'll have to check that out. But... Here is the full empty app interaction.

given:

```
@app
init

@http
get /
options /tasks/:taskOwner
options /tasks/:taskOwner/:taskId
options /tasks

# @aws
# profile default
# region us-west-1
```

```
kenrimple:sandbo/ $ [aws-venv] arc init                                                                    [18:39:02]

     Version ⌁ Architect 6.3.5
         cwd ⌁ /Users/kenrimple/projects/sandbo

✓ Create Existing Architect project manifest (.arc) found
✓ Create Created new project files in src/http
✓ Create Done!
kenrimple:sandbo/ $ [aws-venv] arc deploy                                                                  [18:39:05]
         App ⌁ init
      Region ⌁ us-east-1
     Profile ⌁ krimple
     Version ⌁ Architect 6.3.5
         cwd ⌁ /Users/kenrimple/projects/sandbo

⚬ Deploy Creating new private deployment bucket: init-cfn-deployments-84e0b
⚬ Deploy Initializing deployment
  | Stack ... InitStaging
  | Bucket .. init-cfn-deployments-84e0b
✓ Hydrate Finished checks, nothing to hydrate
⚬ Deploy Created deployment templates
✓ Deploy Generated CloudFormation deployment
✓ Deploy Deployed & built infrastructure
✓ Success! Deployed app in 60.127 seconds

    https://im2sqbb9z1.execute-api.us-east-1.amazonaws.com/staging

kenrimple:sandbo/ $ [aws-venv] arc sandbox                                                               [18:40:11]
         App ⌁ init
      Region ⌁ us-east-1
     Profile ⌁ krimple
     Version ⌁ Architect 6.3.5
         cwd ⌁ /Users/kenrimple/projects/sandbo

× Error: Architect schema validation errors
  @http invalid route: "options /tasks/:taskOwner"
  @http invalid route: "options /tasks/:taskOwner/:taskId"
  @http invalid route: "options /tasks"
kenrimple:sandbo/ $ [aws-venv]```

**Comment by krimple** [comment]
[issue-790.zip](https://github.com/architect/architect/files/4561751/issue-790.zip)

**Comment by ryanblock** [comment]
Yep, this makes sense. As I said (and pointed to in https://github.com/architect/architect/issues/760), `@http options /path` is not currently valid `.arc`

If you were to deploy an Arc file that included such entries in `@http`, that should probably error – but if that didn't error, those `options /whatever` lines would be ignored. By default, right now all `options` requests go to `get /`, as you mentioned.

> Either support in sandbox of lower-level OPTIONS paths or fail deploy...

We won't support something in Sandbox that we don't support in live AWS (for obvious reasons), so taking https://github.com/architect/architect/issues/760 into account – what would you like the outcome to be?

- That we officially support `options` Lambdas on any path, in Sandbox and in live AWS?
- We continue with the `options` going to `get /` (and I guess we error on deploy if we see `options /whatever`)?
- Something else?

**Comment by krimple** [comment]
I'd suggest option #2 as it would match what your expectations currently are for the framework (i.e. the / catchall is where to handle OPTIONS calls). But I don't know if I am really qualified to speak for anyone else using the framework. Thanks for looking at the issue in depth, Ryan.

**Comment by ryanblock** [comment]
Oh, I don't have any particular expectations on this – ultimately we're here to build something people find useful! I opened #760 (option 1) because I thought folks might prefer to have `options` handlers. So, tell you what: I'll make sure behavior is consistent between deploy and sandbox for now, and if you want to ALSO vote in favor of #760, please do that, will take it into account for our roadmap!

**Comment by tbeseda** [comment]
Architect Inventory now handles this and has been refactored; this is likely resolved in subsequent releases. If it is not working then there's definitely a bug. Feel free to create another issue
